app/models/user.rb

```diff
@@ -146,15 +146,16 @@ class User < ApplicationRecord
 
   def expire_caches
     return unless saved_change_to_login?
+    Work.kudosed_by_user(self).touch_all
```

Comment:
I know this can take a long time, I believe this should be moved to an asynchronous task sorry.

Looking at https://apidock.com/rails/ActiveRecord/Relation/touch_all it looks like touch_all changes the record in the database, I don't believe we want to do that we just want to delete the fragments. There are a silly number of Kudo and reseting the username of someone who had kudosed lots of work could create issues. ( If we did have to change the works then it would have to be in a transaction in batches of say 1000 ).

Reply:
Does asynchronous task just mean "a thing that executes asynchronously," or does it have a specific technical meaning in this situation? I was trying to write up a response and then realized that I wasn't sure.

I agree that `touch_all` seems like not quite the right solution, but I'm struggling to figure out how to trigger mass fragment deletion. The key point is that we need to run `ActionController::Base.new.expire_fragment("#{commentable.cache_key}/kudos-v3")` for each work the user has kudosed, right? So is the idea that we somehow trigger an asynchronous task that iterates through a user's kudos (with commentable id) in batches of 1000? Would that be something like a new sweeper that observes users and reacts when they change usernames, or something else?

I notice that `app/sweepers/kudos_sweeper.rb` was removed in #4293 and its functionality moved to `app/models/kudo.rb`; that'll be some fun merging for me, but after looking the change over I don't think it fundamentally alters the core problem/high-level solution here.

Anyway, thanks for the feedback, and sorry it took me a bit to get back to you about it!

Reply:
The key thing here is that to expire the cache, which looks like this:
```
    ActionController::Base.new.expire_fragment("#{kudo.commentable.cache_key}/kudos-v3")
```
we need to expire the fragment for every work a user has kudo'ed (`Work.kudosed_by_user(self)`), either by directly expiring it, or by causing the work's cache_key to change (which touch would do).

I tried looking through the code base to try to find where it would make sense to put an asynchronous task to expire the fragments, and I'm not sure--is it more like indexing, which seems to add things to a queue and then batch that every so often? Or would it be more like one of the rake tasks that performs periodic cleanup? I'm not clear on how to make this work and could use an example.